### ISSUE TITLE
Mac - Read resource file

### ISSUE THREAD

**Issue by LaraSQP** [picoe/Eto#1629]
My app includes a number of text and binary files that the program reads.

This is trivial to do in Linux and Windows. These files are just there in the same folder as the executable and are accessed via the classes in the IO namespace. Zero problems.

But in Mac, these files need(?) to be added as resources to the app bundle (i.e., zipped up) and I have no idea how Mac deals with them. The only reference I have found is:

https://github.com/picoe/Eto/wiki/Xaml-Tips-and-Tricks

So, how does one go about reading a resource file in Mac with Eto?

Any pointers are appreciated.

**Comment by LaraSQP** [comment]
It just hit me.

If Mac forces to embed the file, one might as well do it properly and for all platforms alike.

A bitch of a compromise.

**Comment by LaraSQP** [comment]
For those as unfamiliar as I am with Mac and having a similar problem.

Turns out `Resources` are not always `Resources` in Mac.

Add a text file with `Build Action` as `Content` and it will be available for reading as if it were in a folder, specifically:

	var path	= EtoEnvironment.GetFolderPath( EtoSpecialFolder.ApplicationResources );
	var file	= Path.Combine( path, "a1.txt" );
	var ls		= File.ReadAllLines( file, Encoding.UTF8 );

It gets better.

The so-called `ApplicationResources` folder can also be written to and, in fact, files can be created in it, for example:

	var path	= EtoEnvironment.GetFolderPath( EtoSpecialFolder.ApplicationResources );
	var file	= Path.Combine( path, "a2.txt" );

	File.WriteAllText( file, "something", Encoding.UTF8 );

In other words, not `Resources` at all for they can be modified at run-time.

**Comment by cwensley** [comment]
@LaraSQP I would avoid modifying anything inside the .app bundle on Mac - that will "corrupt" a properly code signed and notarized app (and is against Apple's recommendations).  

You should use the `EtoSpecialFolder.ApplicationSettings` folder instead to store any files needed by your application when running.

**Comment by LaraSQP** [comment]
I can see how it makes sense to use `EtoSpecialFolder.ApplicationSettings`.

However.

The `EtoSpecialFolder.ApplicationSettings` returns the following for Windows, Linux, and Mac respectively:

> C:\Users\user1\AppData\Roaming
> /home/user1/.config
> /Users/user1/Library/Application Support/com.example.FileTest

While the `EtoSpecialFolder.ApplicationResources` returns the following, again, respectively:

> d:\FileTest\FileTest.Wpf\bin\Release\netcoreapp3.1
> /home/user1/Desktop/FileTest
> /Users/user1/Downloads/FileTest.app/Contents/Resources

In other words, `EtoSpecialFolder.ApplicationResources` returns the path to the `Resources` folder in Mac and the path to the executable in Windows and Linux.

On the other hand, `EtoSpecialFolder.ApplicationSettings` returns paths to elsewhere.

Since I am going for portable here (whatever that means in Mac), it seems that `EtoSpecialFolder.ApplicationResources` is the way to go.

I am still puzzled that Mac allows writing to that folder but I know nothing about Mac except that it seems to be doing some funky shit on unzipping the bundle.
